hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestTableLayoutOptimization.scala

```diff
@@ -68,8 +66,8 @@ class TestTableLayoutOptimization extends HoodieClientTestBase {
   }
 
   @ParameterizedTest
-  @ValueSource(strings = Array("COPY_ON_WRITE", "MERGE_ON_READ"))
-  def testOptimizeWithClustering(tableType: String): Unit = {
+  @CsvSource(Array("COPY_ON_WRITE, hilbert", "COPY_ON_WRITE, z-order", "MERGE_ON_READ, hilbert", "MERGE_ON_READ, z-order"))
```

Comment:
Instead of composing this product manually, please use `@MethodSource` and compose it through 2 parameters

Reply:
ok， will  do it

Reply:
@alexeykudinkin  could you pls help me， i donnot know how to use @MethodSource in scala。 thanks